crates/compiler/builtins/roc/Num.roc

```diff
@@ -867,6 +868,35 @@ max = \a, b ->
     else
         b
 
+## Restricts a number to be between two other numbers.
+##
+## ```
+## restrictToInterval 5 3 8 == 5
+## restrictToInterval 1 3 8 == 3
+## restrictToInterval 10 3 8 == 8
+##
+## restrictToInterval 5 8 3 == 5
+## restrictToInterval 1 8 3 == 3
+## restrictToInterval 10 8 3 == 8
+## ```
+restrictToInterval : Num a, Num a, Num a -> Num a
+restrictToInterval = \x, low, high ->
+    if high > low then
+        restrictToInterval x high low
+    else if x < low then
+        low
+    else if x > high then
+        high
+    else
+        x
+
+expect restrictToInterval 5 3 8 == 5
+expect restrictToInterval 1 3 8 == 3
+expect restrictToInterval 10 3 8 == 8
+expect restrictToInterval 5 8 3 == 5
+expect restrictToInterval 1 8 3 == 3
+expect restrictToInterval 10 8 3 == 8
```

Comment:
I think it would be good to test with negative numbers as well.

Reply:
Done as well